### PR TITLE
Run end-to-end tests in Travis CI with CentOS 7 metadata

### PR DESCRIPTION
**Pull request by atodorov**
@henrywang @bcl @wgwoods please review. Currently blocked by https://github.com/weldr/welder-deployment/pull/9

### CODE REVIEW

**Comment by atodorov** [comment]
[Document build/test environment](https://trello.com/c/k92wm2JC/290-document-build-test-environment)

**Comment by atodorov** [comment]
[[QE][UI][Automation] Move end-to-end test into Jenkins CI.](https://trello.com/c/lqzkpCQQ/282-qe-ui-automation-move-end-to-end-test-into-jenkins-ci)

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	npm run test
+
+end-to-end-test:
+	git clone https://github.com/weldr/welder-deployment
+	make -C welder-deployment/ weld-f25
+
+	sudo docker build -t weld/end-to-end:latest .
+
+	make -C welder-deployment/ repos
+	sudo docker-compose build -f welder-deployment/docker-compose.yml
```
Comment (henrywang) [inline Makefile]: Should be sudo docker-compose -f welder-deployment/docker-compose.yml build

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	wget https://s3.amazonaws.com/atodorov/metadata_centos7.db.gz -O welder-deployment/metadata.db.gz
+	gunzip ./welder-deployment/metadata.db.gz
+	make -C welder-deployment/ import-metadata
+	sudo docker-compose -p welder up -d -f welder-deployment/docker-compose.yml
```
Comment (henrywang) [inline Makefile]: Should be `sudo docker-compose -f welder-deployment/docker-compose.yml  -p welder up -d`

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+	npm run lint
+
+test:
+	npm run test
+
+end-to-end-test:
+	git clone https://github.com/weldr/welder-deployment
+	make -C welder-deployment/ weld-f25
+
+	sudo docker build -t weld/end-to-end:latest .
```
Comment (henrywang) [inline Makefile]: This one will build welder-web, not end-to-end because the current folder is the welder-web.

**Comment by atodorov** [comment]
@henrywang thanks for the suggestions. I got this working. See raw log from 
https://travis-ci.org/weldr/welder-web/jobs/225990140

Log:
https://api.travis-ci.org/jobs/225990140/log.txt?deansi=true

End-to-end test were able to execute and they failed with:

```
  18 passing (41s)
  1 failing

  1) Edit Recipe Page Single Word Recipe Name Scenario Create Composition Test should have toast notification pop up when new composition added:
     Error: Timeout of 15000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```

Please merge this PR if no other comments on it.

**Comment by atodorov** [comment]
[(8) [QE] - add sanity tests for docker images](https://trello.com/c/P0VaqlU7/229-8-qe-add-sanity-tests-for-docker-images)

**Pull request merged**